core-processor/src/ext.rs

```diff
@@ -535,11 +535,13 @@ impl Ext {
     }
 
     fn reduce_gas(&mut self, gas_limit: GasLimit) -> Result<(), FallibleExtError> {
-        if self.context.gas_counter.reduce(gas_limit) != ChargeResult::Enough {
-            Err(FallibleExecutionError::NotEnoughGas.into())
-        } else {
-            Ok(())
-        }
+        self.atomic_gas(|ext| {
```

Comment:
mutators for gas and value doesn't require atomic gas

Reply:
https://github.com/gear-tech/gear/pull/3632#pullrequestreview-1809310076 this mostly was about ext trait impls